torch/_export/passes/replace_view_ops_with_view_copy_ops_pass.py

```diff
@@ -0,0 +1,59 @@
+from typing import Dict, List, Set, Type
+
+import torch
+from torch._ops import OpOverload, OpOverloadPacket
+from torch._export.pass_base import ExportPassBase
+
+
+__all__ = ["ReplaceViewOpsWithViewCopyOpsPass"]
+
+
+_NON_FUNCTIONAL_OPS_TO_FUNCTIONAL_OPS: Dict[OpOverload, OpOverload] = {
+    torch.ops.aten._unsafe_view.default: torch.ops.aten.view_copy.default,
+}
+
+# TODO (tmanlaibaatar) remove this after https://github.com/pytorch/pytorch/pull/100749
+_BLACK_LISTED_OPS: Set[OpOverloadPacket] = {
+    torch.ops.aten.sym_size,
+    torch.ops.aten.sym_stride,
+    torch.ops.aten.sym_numel,
+}
+
+def is_view_op(schema: torch._C.FunctionSchema) -> bool:
+    if len(schema.arguments) == 0:
+        return False
+    return schema.arguments[0].alias_info is not None
+
+
+class ReplaceViewOpsWithViewCopyOpsPass(ExportPassBase):
+    """
+    Our backend expects pure functional operators. For efficiency
+    purposes, we keep view ops around while functionalizing the exported
+    program. This pass replaces view ops with view copy ops for backends that
+    need AOT memory planning.
+    """
+    def call_operator(self, op, args, kwargs, meta):
+        if op in _NON_FUNCTIONAL_OPS_TO_FUNCTIONAL_OPS:
+            return super().call_operator(
+                (_NON_FUNCTIONAL_OPS_TO_FUNCTIONAL_OPS[op]), args, kwargs, meta
+            )
+
+        if op in _BLACK_LISTED_OPS:
+            return super().call_operator(op, args, kwargs, meta)
+
+        if op._schema.name.startswith("aten::") and is_view_op(
```

Comment:
can we just do an operator matching like _unsafe_view -> view_copy mapping? It maybe more readable and maintainable.

Reply:
There are many operators that would need to be added to the map. I think it is better to construct it programmatically.

Reply:
How many operators will the contain? The string manipulation logic looks kind of fragile to me. It would be great if we can have some tests to make sure the we can produce proper copy variants for the overloads?

Reply:
I think around 20~, i am not sure. This also has added benefit of detecting any new view/view_copy pairs that might be added in the future. I can guard 
`view_copy_op = getattr(getattr(torch.ops.aten, view_copy_op_name), view_op_overload)` as a precaution. Do you have suggestions for other tests I can do?